### PR TITLE
Feature: Frontend login/signup #3

### PR DESCRIPTION
**Pull request by mcgraths7**
**What it does**

1. Added Login and Signup Routes as well as a persistent Header

2. Both forms were similarly styled, so I added a formStyles file to import from within the 'themes' directory

3. I utilized 'react-hook-form' because it's lightweight compared to other form management libraries

4. Form inputs are validated onBlur, rather than onSubmit for quicker feedback, using helperText from Material UI

**What to look for**

1. I added the 'components', 'hooks', and 'utils' directories (I ended up not using a custom hook, so the hooks directory may or may not be needed). 

2. It's not currently in use, but I created a UserContext object which wraps the whole app so we can keep track of the currently logged in user

3. I'm utilizing a modified version of the Airbnb style guide, I didn't commit my eslint config in case anyone has conflicting rules, but we might want to come to an agreement on this part at some point

4. Navigating to '/' redirects to the signup page, using 'react-router-dom's 'Redirect' component

**What to avoid**

1. I'm not currently validating the login form, as I don't have any accounts to check against

2. The header currently does not switch as it should - need to create a user on the backend first

**Screenshots**
Signup form w/out validation

Signup form w/ validation

Login form

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -5,40 +5,21 @@
...
-    <!--
-      manifest.json provides metadata used when your web app is installed on a
-      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
-    -->
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
 
     <link
       rel="stylesheet"
       type="text/css"
       href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons"
```
Comment (skassam21) [inline client/public/index.html]: Do we still need Roboto if we are using Open Sans?

File: client/src/components/ErrorSnackbar.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    open: false,
+    vertical: 'bottom',
+    horizontal: 'center',
+  });
+
+  const { vertical, horizontal, open } = state;
+
+  // const handleClick = (newState) => () => {
+  //   setState({ open: true, ...newState });
+  // };
```
Comment (skassam21) [inline client/src/components/ErrorSnackbar.jsx]: Try not to commit any commented out code in your PRs (generally bad practice)

File: client/src/themes/theme.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { createMuiTheme } from '@material-ui/core';
 
 export const theme = createMuiTheme({
   typography: {
-    fontFamily: '"Roboto"',
+    fontFamily: '"Open Sans", sans-serif',
   },
   primary: '#f04040',
   secondary: '#1f1f1f',
```
Comment (skassam21) [inline client/src/themes/theme.js]: Can you change the colors in the theme.js as well, so when a TextField is highlighted, it uses our spec's green colors?

File: client/src/pages/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                  type="submit"
+                  variant="contained"
+                  fullWidth
+                  color="primary"
+                  className={classes.action}
+                >
+                  Create
+                </Button>
+              </Grid>
+              <Grid item xs={4} />
```
Comment (skassam21) [inline client/src/pages/Signup.jsx]: I don't like these empty grids to center data - instead you should use the Grid flex-box abilities - you can add justify-content, etc. - read more about it here - https://material-ui.com/components/grid/

File: client/src/pages/Ping.js [hunk 5]
```diff
@@ -1,51 +1,49 @@
-import { FormControl, TextField, Button, Typography } from "@material-ui/core";
+import { FormControl, TextField, Button, Typography } from '@material-ui/core';
```
Comment (skassam21) [inline client/src/pages/Ping.js]: We don't need these components anymore (Ping, etc.)

File: client/src/pages/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import { Link } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+
+import formStyles from '../themes/formStyles';
+// import UserContext from '../components/UserContext';
```
Comment (skassam21) [inline client/src/pages/Login.jsx]: remove commented out code

**Review by skassam21 — COMMENTED** [review]
Nice structure of reusable components + pages. Can you add some details in the root level Readme on how to run the front-end and the front-end's directory structure 

One comment on the directory structure - can we rename the themes folder to like an assets folder, and then we can add imgs, the theme.js file, and any styles in there?

File: client/public/index.html (hunk 1 above)
Comment (mcgraths7) [reply client/public/index.html]: Removed dependency on Roboto from index

File: client/src/components/ErrorSnackbar.jsx (hunk 2 above)
Comment (mcgraths7) [reply client/src/components/ErrorSnackbar.jsx]: Removed this component as I'm no longer using it

File: client/src/themes/theme.js (hunk 3 above)
Comment (mcgraths7) [reply client/src/themes/theme.js]: Will work on this

File: client/src/pages/Signup.jsx (hunk 4 above)
Comment (mcgraths7) [reply client/src/pages/Signup.jsx]: Will work on this

File: client/src/pages/Ping.js (hunk 5 above)
Comment (mcgraths7) [reply client/src/pages/Ping.js]: Will work on this next

File: client/src/pages/Login.jsx (hunk 6 above)
Comment (mcgraths7) [reply client/src/pages/Login.jsx]: Will remove

**Review by skassam21 — APPROVED** [review]
Looks good! Maybe get one more set of eyes on your team to look through it so that they are aware of what's going on here

**Review by cpascale43 — APPROVED** [review]
Looks good!

**Review by Patton-L — COMMENTED** [review]
Looks good

**Pull request merged**